v0/destinations/facebook_pixel/utils.js

```diff
@@ -214,31 +214,34 @@ const transformedPayloadData = (
   return customData;
 };
 
-
 /**
- * 
- * @param {*} message 
+ *
+ * @param {*} message
  * @returns fbc parameter which is a combined string of the parameters below
- * 
+ *
  * version : "fb" (default)
- * 
+ *
  * subdomainIndex : 1 ( recommended by facebook, as well as our JS SDK sets cookies on the main domain, i.e "facebook.com")
- * 
+ *
  * creationTime : mapped to originalTimestamp converted in miliseconds
- * 
+ *
  * fbclid : deduced query paramter from context.page.url
- * 
+ *
  * ref: https://developers.facebook.com/docs/marketing-api/conversions-api/parameters/fbp-and-fbc#fbc
  */
-const deduceFbcParam = (message) => {
-
+const deduceFbcParam = message => {
   const url = message.context?.page?.url;
   if (!url) {
     return undefined;
   }
-  let parseUrl = new URL(url)
-  let paramsList = new URLSearchParams(parseUrl.search);
-  const fbclid = paramsList.get('fbclid')
+  let parseUrl;
+  try {
+    parseUrl = new URL(url);
+  } catch {
+    throw new CustomError("url is not in correct format", 400);
```

Comment:
Why throw error? From that logic, it doesn't seem like it's required that URL is valid.